src/server/game/Warden/WardenCheckMgr.cpp

```diff
@@ -67,73 +62,110 @@ void WardenCheckMgr::LoadWardenChecks()
         }
 
-        if (checkType == MEM_CHECK || checkType == MODULE_CHECK)
-            MemChecksIdPool.push_back(id);
-        else
-            OtherChecksIdPool.push_back(id);
-
         if (checkType == MEM_CHECK || checkType == PAGE_CHECK_A || checkType == PAGE_CHECK_B || checkType == PROC_CHECK)
         {
-            wardenCheck->Address = address;
-            wardenCheck->Length = length;
+            wardenCheck.Address = address;
+            wardenCheck.Length = length;
         }
 
         // PROC_CHECK support missing
-        if (checkType == MEM_CHECK || checkType == MPQ_CHECK || checkType == LUA_STR_CHECK || checkType == DRIVER_CHECK || checkType == MODULE_CHECK)
-            wardenCheck->Str = str;
-
-        CheckStore[id] = wardenCheck;
+        if (checkType == MEM_CHECK || checkType == MPQ_CHECK || checkType == LUA_EVAL_CHECK || checkType == DRIVER_CHECK || checkType == MODULE_CHECK)
+        {
+            wardenCheck.Str = str;
+        }
 
         if (checkType == MPQ_CHECK || checkType == MEM_CHECK)
         {
-            WardenCheckResult* wr = new WardenCheckResult();
-            wr->Result.SetHexStr(checkResult.c_str());
-            int len = checkResult.size() / 2;
-            if (wr->Result.GetNumBytes() < len)
+            WardenCheckResult wr;
+            wr.Result.SetHexStr(checkResult.c_str());
+            int len = static_cast<int>(checkResult.size()) / 2;
+            if (wr.Result.GetNumBytes() < len)
             {
                 uint8* temp = new uint8[len];
                 memset(temp, 0, len);
-                memcpy(temp, wr->Result.AsByteArray().get(), wr->Result.GetNumBytes());
+                memcpy(temp, wr.Result.AsByteArray().get(), wr.Result.GetNumBytes());
                 std::reverse(temp, temp + len);
-                wr->Result.SetBinary((uint8*)temp, len);
+                wr.Result.SetBinary((uint8*)temp, len);
                 delete [] temp;
             }
             CheckResultStore[id] = wr;
         }
 
         if (comment.empty())
-            wardenCheck->Comment = "Undocumented Check";
+            wardenCheck.Comment = "Undocumented Check";
         else
-            wardenCheck->Comment = comment;
+            wardenCheck.Comment = comment;
+
+        // Prepare check pools
+        switch (checkType)
+        {
+            case MEM_CHECK:
+            case MODULE_CHECK:
+            {
+                CheckIdPool[WARDEN_CHECK_MEM_TYPE].push_back(id);
+                break;
+            }
+            case LUA_EVAL_CHECK:
+            {
+                if (wardenCheck.Length > WARDEN_MAX_LUA_CHECK_LENGTH)
+                {
+                    sLog->outError("sql.sql: Found over-long Lua check for Warden check with id %u in `warden_checks`. Max length is %u. Skipped.", id, WARDEN_MAX_LUA_CHECK_LENGTH);
+                    continue;
+                }
+
+                std::string str = fmt::sprintf("%04u", id);
```

Comment:
![image](https://user-images.githubusercontent.com/24550914/104850234-27e96500-58ee-11eb-9b9e-1d1e6959de94.png)

According to Codefactor this is still an issue https://www.codefactor.io/repository/github/azerothcore/azerothcore-wotlk/pull/3875

Reply:
i cannot solve which doesnt exists

Reply:
https://github.com/sanctum32/azerothcore-wotlk/blob/warden_pqr/src/server/game/Warden/WardenCheckMgr.cpp#L141

Reply:
I dont know if it can be left or needs to be resolved as I don't have the knowledge of what overflows are yet

Reply:
I dont know what you mean

```cpp
std::string str = fmt::sprintf("%04u", id);
```
is present in the PR, the link above as well as codefactor :shrug:

Reply:
Maybe @Winfidonarleyan knows?

Reply:
anyways i done with my prs, i don't know why codefactor thinks that fmt::printf is used while there fmt::sprintf is used.

Reply:
it doesnt think fmt::printf is used, it is complaining about sprintf

Reply:
But it is up to @FrancescoBorzi if it can be ignored and it sounded on @Winfidonarleyan he knew how to fix it

Reply:
"acore::StringFormat" call would give same.

Reply:
@Kitzunu its pretty much false report, look, https://github.com/sanctum32/azerothcore-wotlk/blob/warden_pqr/src/server/game/Warden/WardenCheckMgr.cpp#L66 check for lua is done here and that fmt::sprintf tried to access 4 symbol number, so maximum value should be 9999

Reply:
ok, trusting you as Shin put it to be merge earlier too